app.js

```diff
@@ -1,52 +1,68 @@
-$(document).ready((e) => {
+$(document).ready(() => {
+    (function ($) {
 
-    const key = myL3.apiKeys.openweathermapKey;
+        fluid.registerNamespace("myL3");
+        
+        myL3.TABLE_PARAMS = [   // Parameters in table.
```

Comment:
`myL3.TABLE_PARAMS` appears to only be used to build the HTML table in `myL3.weatherService.addDataIntoTable`. We should define it closer to that file. A better name is probably also something like "weatherTableLabels" or similar. And I think we should also change the structure to key-string lookup:

```
{
    tempF: "Temperature (F)",
    tempC: "Temperature (C)",
    ...
}
```

Making the strings an array is brittle -- it is tied (though it's not clear that it is) to the order of the checkboxes in the UI and will break if that UI is changed.

I will add a similar comment regarding the `fields` parameter of `myL3.weatherService.addDataIntoTable`

Reply:
Earlier I tried it using key-value pairs but that wasn't looking efficient. In the following loop, If we'll use key-value pairs then we'll have to check all the time, which value corresponds to which field in weather table (using switch case or if-else-if). This will make our code longer and less efficient. So the way I chose to implement is to store the labels in the array, in the order they are defined in html i.e. their values will be one more than the index of array. As you said it looks brittle but if we create checkboxes in modal and weatherTableLabels[] array elements in same order then this will work perfectly. Meanwhile, I am putting necessary comments in the code to make it clear to everyone reading it in future.
```    
for(let i=0; i<myL3.values.length; i++) {
        $('#tabularData> tbody > tr:last').after("<tr><td>" + weatherTableLabels[parseInt(fields[i])-1] + 
        "</td><td>" + myL3.values[i] + "</td></tr>");
    }
```